doc/api/diagnostics_channel.md

```diff
@@ -407,6 +457,526 @@ channel.subscribe(onMessage);
 channel.unsubscribe(onMessage);
 ```
 
+#### `channel.bindStore(store[, transform])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `store` {AsyncLocalStorage} The store to which to bind the event data
+* `transform` {Function} Transform the event data before setting the context
+
+When [`channel.runStores(data, ...)`][] is called, the given event data will
```

Comment:
I might be missing context, but I don't understand why this function is useful. Could you elaborate?

Reply:
You mean `channel.runStores(...)`? It exists to allow for mapping diagnostics_channel event data to an AsyncLocalStorage instance. The main use case is for APMs to use event data to create a span around some function call and any descending async behaviour.

Reply:
Sorry, I meant `channel.bindStore`

Reply:
The `bindStore` method is how an `AsyncLocalStorage` instance is connected to a given event. It's the `AsyncLocalStorage` equivalent of the `subscribe` method. When a store is bound to a channel it will set its context to the event data whenever the event occurs, for the scope of the function given to the `runStores(...)` method. If a `transform` function was given with in the `bindStore` call it will be used to transform the event data before it becomes the context value for that particular store.

For example, an APM might want to do something like:

```
const mysqlTrace = dc.tracingChannel('mysql.query')

// In APM code somewhere...
mysqlTrace.bindStore(apmStore, (queryData) => {
  const parent = apmStore.getStore()
  return new Span('mysql.query', {
    query: queryData.query
  }, parent)
})

// In mysql library...
class Connection {
  query(queryString, values, callback) {
    mysqlTrace.traceCallback(function wrappedQuery (queryString, values, callback) {
      // original query code goes here...
    }, 2, {
      query: queryString
    }, this, queryString, values, callback)
  }
}
```

When that is done, the `Span` object returned by the `transform` function will become the current context within the scope of that wrappedQuery function and any nested async activity. Until it gets set to something else. APMs generally will chain the spans together by linked to a "parent" span which is what the `const parent = apmStore.getStore()` part represents.

Reply:
Thanks for the explanation!